[Doc Feature] [Meeting apps support in Outlook preview][3291902] (#9321)
MicrosoftDocs/msteams-docs: msteams-platform/m365-apps/extend-m365-meeting-app.md

    "manifestVersion" : "1.16"
}
```

### Use TeamsJS version 2.5.0 or later

Use TeamsJS client library version 2.5.0 or later to extend your Teams meeting app to run in Outlook. If your app uses TeamsJS v2.5.0 or later, application code changes aren't required to run in Outlook.

If you need to upgrade your TeamsJS npm package from v1.x.x to the latest v2.x.x TeamsJS, you can use Teams Toolkit to help identify and automate the required code changes. For more information, see [TeamsJS library](../tabs/how-to/using-teams-client-library.md#whats-new-in-teamsjs-version-2xx).

Following are the steps to update your TeamsJS to the latest version using Teams toolkit:

1. Open your app in Visual Studio Code and select **Teams Toolkit** extension.
1. Under **View**, select **Command Palette...** or **Ctrl+Shift+P**.
1. Run the command `Teams: Upgrade Teams JS SDK and code references`.

Your package.json file references `@microsoft/teams-js@2.x.x` and your `*.js/.ts` and `*.jsx/.tsx` files are updated with:

* Import statements for `teams-js@2.x.x`.
* [Function, Enum, and Interface calls](../tabs/how-to/using-teams-client-library.md#whats-new-in-teamsjs-version-2xx) for `teams-js@2.x.x`.

> * `TODO` comment reminders flagging areas that might be impacted by [Context](../tabs/how-to/using-teams-client-library.md#updates-to-the-context-interface) interface changes.

Reviewer: Update: apologies, I saw the context in the Teams channel post thread

I think this may lead to some confusion though... perhaps quotations can help communicate, please let me know what you think:
```suggestion
* New "`TODO`" comment reminders flagging areas that might be impacted by [Context](../tabs/how-to/using-teams-client-library.md#updates-to-the-context-interface) interface changes.
* New "`TODO`" comment reminders to [convert callback functions to promises](../tabs/how-to/using-teams-client-library.md#callbacks-converted-to-promises).
```

Follow these steps to update your TeamsJS to the latest version using Teams Toolkit:

1. Open your app in Visual Studio Code and select **Teams Toolkit** extension.
1. Under **View**, select **Command Palette...** or **Ctrl+Shift+P**.
1. Run the command `Teams: Upgrade Teams JS SDK and code references`.

Your package.json file references `@microsoft/teams-js@2.x.x` and your `*.js/.ts` and `*.jsx/.tsx` files are updated with:

* Import statements for `teams-js@2.x.x`.
* [Function, Enum, and Interface calls](../tabs/how-to/using-teams-client-library.md#whats-new-in-teamsjs-version-2xx) for `teams-js@2.x.x`.
* New "`TODO`" comment reminders flagging areas that might be impacted by [Context](../tabs/how-to/using-teams-client-library.md#updates-to-the-context-interface) interface changes.
* New "`TODO`" comment reminders to [convert callback functions to promises](../tabs/how-to/using-teams-client-library.md#callbacks-converted-to-promises).

> * `TODO` comment reminders to [convert callback functions to promises](../tabs/how-to/using-teams-client-library.md#callbacks-converted-to-promises).

Reviewer: My suggestion accidentally caused this duplicate line, @erikadoyle , could you please remove the extra line 115?

Reviewer 2: Updated

---
title: Extend a Teams meeting app to Outlook
description: Here's how to ensure your Teams meeting app experience works seamlessly when extended to Outlook
ms.date: 08/21/2023
ms.author: mosdevdocs
author: erikadoyle
ms.topic: tutorial
ms.localizationpriority: medium
ms.subservice: m365apps
---
# Extend a Teams meeting app to Outlook

> [!NOTE]
>
> * While in preview, Outlook support for Teams meeting apps is only available in [Beta Channel builds](https://insider.microsoft365.com/en-us/join/windows) of the Outlook for Windows desktop client.

> 

Author: Thanks, removed the asterisk. I don't think the Teams developer preview note applies here, as this is a Teams GA feature that is preview only in Outlook.

The following table shows the Teams meeting app contexts supported in Outlook:

| Teams meeting context | app manifest value | TeamsJS value | Outlook support
|--|--|--|--|
| [Meeting details view](../apps-in-teams-meetings/build-tabs-for-meeting.md#meeting-details-view) | `meetingDetailsTab` | `frameContext.content` | ✔️ |
| [Meeting chat view](../apps-in-teams-meetings/build-tabs-for-meeting.md#meeting-chat-view) | `meetingChatTab` | `frameContext.content` | ✔️ |
| [Meeting side panel view](../apps-in-teams-meetings/build-tabs-for-meeting.md#meeting-side-panel-view) | `meetingSidePanel` | `frameContext.sidePanel` | ✔️ |
| [Meeting stage view](../apps-in-teams-meetings/build-tabs-for-meeting.md#meeting-stage-view) | `meetingStage` | `frameContext.meetingStage` | ❌ |

If your meeting app supports stage view, other in-meeting effects, or contains elements such as message extensions and bots, these will continue to work in Teams when the meeting is scheduled from Outlook, but will not otherwise surface in Outlook.

> The rest of this guide walks your through the steps to ensure your Teams meeting app experience works seamlessly when extended to Outlook.

Author: The entire article (not just the later part) is about this, so how about:

This article walks your through the steps to ensure your Teams meeting app experience works seamlessly when extended to Outlook.

> * [Update your Microsoft AD App Registration for SSO](#update-azure-ad-app-registration-for-sso).
> * [Sideload your updated app using TeamsFx CLI](#sideload-your-app-using-teamsfx-cli).

If you have an existing meeting app, make a copy or a branch of your production project for testing and update your App ID in the app manifest to use a new identifier (distinct from the production App ID) for testing.

### Use app manifest version 1.13+

Use app manifest schema version `1.13` or higher for extending Teams apps to run across other Microsoft 365 hosts, including Outlook. If you need to upgrade your app manifest from an earlier version, you have two options:

# [Teams Toolkit](#tab/manifest-teams-toolkit)

1. Open the command palette: `Ctrl+Shift+P`.
1. Run the `Teams: Upgrade Teams manifest` command and select your app manifest file. Changes are made in place.

# [Manual steps](#tab/manifest-manual)

Open your Teams app manifest and update the `$schema` and `manifestVersion` manually with the appropriate version. Refer to [Teams developer manifest](../resources/schema/manifest-schema.md) for version information:

```json
{

>     "$schema" : "https://developer.microsoft.com/json-schemas/teams/v1.14/MicrosoftTeams.schema.json",

Author: Good catch!

    "manifestVersion" : "1.16"
}
```

---

### Use TeamsJS version 2.5.0+

Use TeamsJS client library version `2.5.0` or higher to extend your meeting app to run in Outlook. As long as your app uses TeamsJS v2.5.0 or higher, no application code changes are required to run in Outlook.

If you need to upgrade your TeamsJS npm package from v1.x.x to the latest (2.x.x) TeamsJS, you can use Teams Toolkit to help identify and automate the required code changes. Alternately, you can perform the same steps manually; refer to [TeamsJS library](../tabs/how-to/using-teams-client-library.md#whats-new-in-teamsjs-version-2xx) for details.

> 1. Open the *Command palette*: `Ctrl+Shift+P`.

Author: I'm not sure I understand the suggestion. The user doesn't need to switch to the Teams Toolkit tab in VS Code because we're accessing the required Toolkit functionality via global VS Code command palette.

    :::image type="content" source="images/toolkit-zip-teams-app-package.png" alt-text="Screenshot shows the Zip Teams App Package option in Teams Toolkit extension for Visual Studio Code.":::

1. The preview for Teams meeting app support in Outlook requires you to sideload your app package using TeamsFx CLI. (Sideloading from the *Manage your apps* option in Teams client is not currently supported.) Ensure TeamsFx CLI is installed:

    ```bash
    npm install -g @microsoft/teamsfx-cli
    ```

1. Use the following command to sideload your app package to Teams, replacing *<path\to\appPackage.zip>* with the path to your app package.

    ```bash
    teamsfx m365 sideloading --file-path <path\to\appPackage.zip>
    ```

After it's sideloaded to Teams, your meeting app will be available for use in both Teams and Outlook. It may take several minutes and a restart of the Outlook for Windows desktop client for your app to appear in the **Apps** menu.

## Preview your meeting app in Outlook

> Here's how to preview your meeting app running on Outlook for Windows desktop:

Author: "Outlook for Windows desktop" is the correct wording per Office PM

> Here's how to preview your meeting app running on Outlook for Windows desktop:

Author: (Also want to be explicit we're talking about desktop, not web client)

> 1. With your meeting app running (either remotely from Azure, or locally from Teams Toolkit `F5`), open Outlook for Windows desktop signed in to your test tenant account. If Outlook is already running, close and restart it again.

Reviewer 2: ```suggestion
1. Open Outlook for Windows with the meeting app running either remotely from Azure, or locally from Teams Toolkit. If Outlook is already running, close and restart again.
```

Author: I don't think we can reorder the clauses here--the meeting app needs to be running first before opening Outlook.

> 1. Open *Calendar*, then select **New Meeting**. By default, new meetings are *Microsoft Teams* meetings. For a meeting app to run in Outlook, the meeting must be a Teams meeting (meeting apps are not supported in non-Teams meetings scheduled from Outlook).

Reviewer 2: ```suggestion
1. Open **Calendar** and select **New Meeting**. For a meeting app to run in Outlook, the meeting must be a Teams meeting as meeting apps aren't supported in other meetings scheduled from Outlook.
```

> 1. From the meeting scheduling window, select **All Apps**. Your sideloaded meeting app should appear among your installed meeting apps.

Reviewer 2: ```suggestion
1. From the meeting scheduling window, select **All Apps**. Your sideloaded meeting app appears among your installed meeting apps.
```

> :::image type="content" source="images/outlook-meeting-all-apps.png" alt-text="Sideloaded meeting app showing in `All Apps` menu of Outlook meeting scheduler":::

Reviewer 2: Add a line space before image as it's merging with the text.

> 1. Select your meeting app. This opens its configuration dialog, where your user will configure any relevant app settings for their specific meeting.

Reviewer 2: ```suggestion
1. Select your meeting app. This opens a configuration dialog, where your app user will configure any relevant app settings for their specific meeting.
```

> :::image type="content" source="images/outlook-meeting-app-configuration.png" alt-text="Meeting app configuration page showing from Outlook meeting scheduler":::

Reviewer 2: Add a line space before image as it's merging with the text.

> 1. Click **Add** to dismiss the dialog. Your meeting app will display in the side panel of the scheduling window and the **Apps** field will indicate its availability to participants in the meeting. From the **Apps* list, you can open the dropdown menu on your app to see how the app appears in the Store (About), update the app configuration for this meeting (Settings), or remove the app from this meeting (Remove).

Reviewer 2: Please follow the writing guidelines and break it into multiple steps as too many actions are in one step.

>     App developers can configure the appearance of *Settings* and *Remove* in Outlook, the same as with [other configurable tabs in Teams](../tabs/how-to/create-tab-pages/removal-page.md#enable-your-tab-to-be-reconfigured-after-installation).

Reviewer 2: ```suggestion
You can configure the appearance of **Settings** and **Remove** in Outlook, the same as with [other configurable tabs in Teams](../tabs/how-to/create-tab-pages/removal-page.md#enable-your-tab-to-be-reconfigured-after-installation).
```

    :::image type="content" source="images/outlook-meeting-app-side-panel.png" alt-text="Meeting app running in Outlook meeting scheduler side panel":::

> 1. To uninstall the meeting app (so its no longer available from the *All Apps* menu in Outlook), open Teams and select *Apps* > **Manage your apps**. Select your sideloaded meeting app in the list, and then select **Remove**. You will need to restart Outlook to see the change reflected in the *All Apps* menu from the meeting scheduler.

Reviewer 2: ```suggestion
1. To uninstall the meeting app, open Teams and select **Apps** > **Manage your apps**. Select your sideloaded meeting app in the list, and **Remove**. You need to restart Outlook to see the change reflected in the **All Apps** menu from the meeting scheduler.
```

> :::image type="content" source="images/teams-manage-your-apps-remove.png" alt-text=" `Manage your apps` pane of Microsoft Teams showing the sideloaded meeting app `Remove` option":::

Reviewer 2: Add a line space before image as it's merging with the text.

Thanks for previewing Teams meeting app support in Outlook! Please use the [Microsoft Teams developer community channels](/microsoftteams/platform/feedback) to report issues and provide feedback.

## Code sample

| **Sample Name** | **Description** | **Node.js** |
|---------------|--------------|--------|
| HelloWorld In-meeting app| Simple meeting app demonstrating meeting details view, chat view, and side panel. Works in Teams, Outlook. | [View](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/hello-world-in-meeting)  |

## See also

* [Extend Teams apps across Microsoft 365](overview.md)

> * [Public developer preview for Teams](../resources/dev-preview/developer-preview-intro.md)

Reviewer 2: Global comment

Please follow the style guide as there are multiple language and consistency issues.

Also the images are misaligned.

Author: Fixed misalignment and tightened up the language (and Acrolinx score).
